### PR TITLE
Release 2.0.0

### LINKED ISSUE #228: Using systemd, run sidekiq as a system service rather than "user service"
Hi all

Currently this gem invokes `systemctl` hardcoding the `--user` option, like this:

```ruby
execute :systemctl, '--user', 'start', fetch(:service_unit_name)
```

This prevents users from running Sidekiq as a _system service_ rather than as a _user service_. It would be great if one could specify this with a configuration option. Something like:

```ruby
set :init_system_type, :system   # or ':user'
```

What do you think? I could work on a PR for this if you consider it.

--- 

**A bit of context of why I think this could be a good idea:**

We're currently deploying a Rails app with Capistrano. We have two roles: a web server role and a worker role; only the latter will run Sidekiq, along with other services. Instances of both roles run in two different AWS Autoscaling Groups respectively. Autoscaling the worker is a little tricky, because we want it to boot and launch Sidekiq and the other services immediately.

If we configure the services as "`--user`" services, these will only be started when a user starts a session in the machine; but we need the services to start up automatically just after the machine boots.

### PR DESCRIPTION
**Pull request by seuros**
closes #233 #234 
fixes #228

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,11 +19,17 @@ And then execute:
...
 ## Usage
 ```ruby
-# Capfile
-require 'capistrano/sidekiq'
-require 'capistrano/sidekiq/monit' #to require monit tasks # Only for capistrano3
+    # Capfile
+
+    require 'capistrano/sidekiq'
+    install_plugin Capistrano::Sidekiq  # Default sidekiq tasks
+    # Then select your service manager
```
Comment (dmitry) [inline README.md]: Please include monit here.

File: lib/capistrano/tasks/systemd.rake [hunk 2]
```diff
@@ -0,0 +1,92 @@
+git_plugin = self
+
+namespace :sidekiq do
+  desc 'Quiet sidekiq (stop fetching new tasks from Redis)'
+  task :quiet do
+    on roles fetch(:sidekiq_roles) do |role|
+      git_plugin.switch_user(role) do
+        execute :systemctl, "--user", "reload", fetch(:sidekiq_service_unit_name), raise_on_non_zero_exit: false
```
Comment (Tietew) [inline lib/capistrano/tasks/systemd.rake]: `"--user"` still hardcoded against #228.

**Comment by 23tux** [comment]
Is there any news on this PR?

**Comment by cupnoodle** [comment]
Anyway we can help on this PR?

**Comment by egracens** [comment]
Hi @seuros! Hope you are doing fine! Will you finish this PR?

**Comment by seuros** [comment]
@egracens , thank you for the ping. 
I totally forgot about it. 
Yes. i will now.

**Comment by seuros** [comment]
Can you use `sudo` or i do a user witch to root  to install the global systemd service ?

File: README.md (hunk 1 above)
Comment (dmitry) [reply README.md:29]: @seuros please include monit here.

Comment (seuros) [reply README.md:29]: Readme is totaly outdated, i will do that once the gem is full working in this branch.

You can open PR against this branch if you can help. I will merge

**Comment by andrepcg** [comment]
Any news?

**Comment by JoeDupuis** [comment]
What is missing? Can we help? Version 1.0.2 works great for me with sidekiq 6.0.6 using Systemd.

**Comment by andrepcg** [comment]
@JoeDupuis 

```
Bundler could not find compatible versions for gem "sidekiq":
  In snapshot (Gemfile.lock):
    sidekiq (= 6.0.7)

  In Gemfile:
    sidekiq

    capistrano-sidekiq was resolved to 1.0.3, which depends on
      sidekiq (>= 3.4, < 6.0)
```

**Comment by JoeDupuis** [comment]
Version 1.0.3 lock the sidekiq gem to `< 6.0`, but the 1.0.2 doesn't. And it works fine (using systemd) if you want to upgrade to 6 without waiting for capistrano-sidekiq 2.0.

Just lock capistrano-sidekiq to 1.0.2 and you should be fine.

**Comment by 23tux** [comment]
@JoeDupuis thanks for the hint, but when I try to use 1.0.2 and id calls `sidekiq:start` during the deployment, I get this error

```
      01 invalid option: --index
      01 /var/www/app/shared/bundle/ruby/2.6.0/gems/sidekiq-6.0.7/lib/sidekiq/cli.rb:291:in `parse_options'
      01 /var/www/app/shared/bundle/ruby/2.6.0/gems/sidekiq-6.0.7/lib/sidekiq/cli.rb:205:in `setup_options'
      01 /var/www/app/shared/bundle/ruby/2.6.0/gems/sidekiq-6.0.7/lib/sidekiq/cli.rb:24:in `parse'
      01 /var/www/app/shared/bundle/ruby/2.6.0/gems/sidekiq-6.0.7/bin/sidekiq:27:in `<top (required)>'
      01 /var/www/app/shared/bundle/ruby/2.6.0/bin/sidekiq:23:in `load'
      01 /var/www/app/shared/bundle/ruby/2.6.0/bin/sidekiq:23:in `<top (required)>'
```

**Comment by seuros** [comment]
@23tux , master will work for you.
I'm going to test it with a centos instance(i tested only in ubuntu) and release the version

**Comment by seuros** [comment]
Sorry, i meant this branch.

**Comment by bayevels** [comment]
I tested it on a Debian 10 Server worked perfectly Thanks :)

**Comment by seuros** [comment]
Thanks for the feedback.

I'm releasing 2.0.0.beta1 .

I will test in other OSes. 

And we readd features dropped such as multi processes support.

**Comment by ivanovv** [comment]
I'm a bit late for the party with this question, but what was the reason to drop sidekiq 5 support?

**Pull request merged**